### DISCUSSION TITLE
How to use a floating-point numerical tolerance in diff()

### DISCUSSION THREAD

**Discussion by damienhocking** [nlohmann/json#4150] [Q&A] [answered]
I've read the various discussions on floating-point comparisons, for example #3050 and #703 and I've read the comparison section of the docs. If you can't use a custom NumberFloatType with its own equality operator, is there a straightforward way to do a numerical tolerance check with diff()? We use json for results of simulations and when simulations are run on say X64 vs ARM64, there are very small differences that we'd like to ignore. 

We could use flatten() and the equal_within_tolerance solution for #3050 and loop through the whole list of keys, but we'd prefer built-in fucntionality if we could, we've found diff() to be very useful.

Thanks in advance..

**Comment by nlohmann** [comment, marked as answer]
This is currently not possible.

**Reply by damienhocking** [reply]
OK, thanks for the fast response.
